clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3118,6 +3120,76 @@ protected void processVersionSwapMessage(
     }
   }
 
+  @Override
+  protected void processSyncOffsetMessage(
+      ControlMessage controlMessage,
+      int partition,
+      PartitionConsumptionState partitionConsumptionState) {
+    // Update the latest processed upstream offset in memory.
+    OffsetRecord offsetRecord = partitionConsumptionState.getOffsetRecord();
+    PubSubTopic upstreamTopic = offsetRecord.getLeaderTopic(pubSubTopicRepository);
+    if (upstreamTopic != null && upstreamTopic.isRealTime()) {
```

Comment:
Hi @huangminchn, I'm wondering if this upstream topic check is necessary for consumers of the SyncOffset CM.